sdk/core/Azure.Core/src/Internal/ArrayBackedPropertyBag.cs

```diff
@@ -68,7 +68,7 @@ public void GetAt(int index, out TKey key, out TValue value)
             {
                 0 => _first,
                 1 => _second,
-                _ => _rest[index - 2]
+                _ => _rest![index - 2] // Must throw if _rest is null
```

Comment:
Will this throw the desired exception type with a message that helps the customer understand how to address the issue?